### DISCUSSION TITLE
What is the relationship between `lsp-config` and `nvim-cmp`

### DISCUSSION THREAD

**Discussion by jinleileiking** [hrsh7th/nvim-cmp#1581] [Q&A] [answered]
I see codes in example:

```
    486 │ -- Setup lspconfig.¬
  ┃  487 │ -- local capabilities = require('cmp_nvim_lsp').default_capabilities(vim.lsp.protocol.make_client_capabilities())¬
  ┃  488 │ -- -- Replace <YOUR_LSP_SERVER> with each lsp server you've enabled.¬
  ┃  489 │ -- require('lspconfig')['<YOUR_LSP_SERVER>'].setup {¬
  ┃  490 │ --   capabilities = capabilities¬
  ┃  491 │ -- }¬

```

**Comment by hrsh7th** [comment, marked as answer]
1. The `Language Server Protocol` has `capbilities` concept.
2. The neovim does not have completion related features in built.
3. The nvim-cmp provides completion related features.
4. The `cmp_nvim_lsp` source supports `Language Server Protocol`.
5. So `cmp_nvim_lsp` tells `I supports completion related features` to the language servers.
